sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -138,7 +135,6 @@ def set_params(self, **kwargs):
             Estimator instance.
         """
         super().set_params(**kwargs)
-        self._validate_params()
         return self
```

Comment:
You can just remove this `BaseSGD.set_params` method and just inherit from the parent class (`BaseEstimator`).